server/cshr/views/vacations.py

```diff
@@ -189,12 +189,14 @@ def post(self, request: Request) -> Response:
             applying_user = request.user
 
             # Check if there are pending vacations in the same day.
-            pending_requests = Vacation.objects.filter(
-                from_date__day=start_date.day,
-                end_date__day=end_date.day,
-                status=STATUS_CHOICES.PENDING,
-                applying_user=applying_user,
-            )
+            date_range = [start_date + timedelta(days=x) for x in range((end_date - start_date).days + 1)]
+            for date in date_range:
+                pending_requests = Vacation.objects.filter(
+                    from_date__lte=date,
+                    end_date__gte=date,
+                    status=STATUS_CHOICES.PENDING,
+                    applying_user=applying_user,
+                )
```

Comment:
You don't need to do for loop on a database transaction, you could easily filter based on your dates

```suggestion
                pending_requests = Vacation.objects.filter(
                    from_date__lte__in=date_range,
                    end_date__gte__in=date_range,
                    status=STATUS_CHOICES.PENDING,
                    applying_user=applying_user,
                )
```

Reply:
The suggested code gave me an error. I removed the loop and changed the query to check for overlapping dates to get the range.